low-level/files/opt/seagate/sspl/setup/resource_map/server.py

```diff
@@ -234,4 +234,38 @@ def prepare_mem_json(self, status, description):
         self.set_health_data(
             memory_dict, status=status, description=description,
             specifics=specifics)
-        return memory_dict
\ No newline at end of file
+        return memory_dict
+
+    def get_fanmodules_info(self):
+        """Get the Fan sensor information using ipmitool."""
+        data = []
+        sensor_reading = self._ipmi.get_sensor_list_by_type('Fan')
+        for fan_reading in sensor_reading:
+            sensor_id = fan_reading[0]
+            fan_dict = self.get_health_template(sensor_id, is_fru=True)
+            fan_dict["last_updated"] = int(time.time())
+            health_desc = "good"
+            recommendation = "NA"
+            sensor_props = self._ipmi.get_sensor_props(sensor_id)
+            status = 'OK' if fan_reading[2] == 'ok' else 'NA'
+            if status != 'OK':
+                health_desc = "bad"
+                recommendation = DEFAULT_RECOMMENDATION
+            description = f"{sensor_id} is in {health_desc} health"
```

Comment:
No need to define 'description' here, set_health_data function takes care of adding description field. you can remove this line and with this change no need to pass 'description' parameter to set_health_data function -line 265.

Reply:
Ack,
Removed the description & recommendation fields